### ISSUE TITLE
`no-extra-semicolons` and `declaration-empty-line-before` false positive for SCSS nested properties

### ISSUE THREAD

**Issue by aweiher** [stylelint/stylelint#2456]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I get false positives when I use SCSS nested properties.

> Which rule, if any, is this issue related to?

- `no-extra-semicolons`
- `declaration-empty-line-before`

> What CSS is needed to reproduce this issue?

**minimal-testcase.scss**
```scss
.some-rule {
    display: inline-block;
    padding: {
        top: 3px;
        right: 6px;
        bottom: 3px;
        left: 7px;
    };
    text-transform: uppercase;
}
```

> What stylelint configuration is needed to reproduce this issue?

**minimal-stylelint.js**
```json
module.exports = {
  "extends": ["stylelint-config-standard"]
}
```

> Which version of stylelint are you using?

`7.9.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`./node_modules/.bin/stylelint --config minimal-stylelint.js minimal-testcase.scss`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS nested properties

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

I get the following warnings: 

```shell

minimal-testcase.scss
 8:4  ✖  Unexpected extra semicolon               no-extra-semicolons          
 9:3  ✖  Expected empty line before declaration   declaration-empty-line-before

```

> Additional Informations

I reported this issue already for `stylelint-order` ( https://github.com/hudochenkov/stylelint-order/issues/20 ) which was resolved by @hudochenkov in this commit: https://github.com/hudochenkov/stylelint-order/commit/2d1757d959f0ef0ab1f391aa3dac6197e5c540f6

I guess the fix for the other rules goes in the same direction?

**Comment by evilebottnawi** [comment]
@aweiher because it is extra semicolon
try to write
```scss
.some-rule {
    display: inline-block;
    padding: {
        top: 3px;
        right: 6px;
        bottom: 3px;
        left: 7px;
    }
    text-transform: uppercase;
}
```

**Comment by evilebottnawi** [comment]
@aweiher see examples http://sass-lang.com/documentation/file.SASS_REFERENCE.html#nested_properties

**Comment by hudochenkov** [comment]
The problems is `postcss-scss` parses `a { margin: { left: 10px; }}` as a rule with selector `margin:`, but `a { margin: 0 { left: 10px; }}` parsed as a declaration with a prop `margin`.

This is the reason behind it:

> ```
> margin: {
>   top: 10px
> }
> ```
> should be parsed as Rule, because `postcss-nested-props` plugin expect it.

> [Source](https://github.com/postcss/postcss-scss/pull/40#issuecomment-242723342)

And I agree with [what Andrey Sitnik said](https://github.com/postcss/postcss-scss/pull/40#issuecomment-239150557):

> I think that this feature is too rare even for Sass users.

I had doubts if it worth to fix in `stylelint-order`. Decided to fix, but it made the code more complicated.

**Comment by aweiher** [comment]
Ok, thank you for your input. No problem after all, as I already removed the semicolons in my codebase, and will refactor nested scss properties also now. Thank you for your time :)

**Comment by aweiher** [comment]
For the future readers, consider to disallow nested properties in your scss with this rule: 

https://github.com/kristerkari/stylelint-scss/blob/master/src/rules/declaration-nested-properties/README.md
